packages/ts/react-grid/src/date-formatter.ts

```diff
@@ -0,0 +1,54 @@
+function getFormatRegex(format: Intl.DateTimeFormat) {
+  const sampleDate = new Date(1234, 4, 6);
+  const formattedSample = format.format(sampleDate);
+  const pattern = formattedSample
+    .replace('1234', '(?<year>\\d+)')
+    .replace('05', '(?<month>\\d+)')
+    .replace('5', '(?<month>\\d+)')
+    .replace('06', '(?<day>\\d+)')
+    .replace('6', '(?<day>\\d+)');
+
+  return new RegExp(pattern, 'u');
+}
+
+export interface DateObject {
+  year: number;
+  month: number;
+  day: number;
+}
```

Comment:
This is what the date picker API uses, so I built the formatter class around that rather than using actual date instances.